commit/plugin_functions.go

```diff
@@ -585,6 +585,50 @@ func validateObservedGasPrices(gasPrices []cciptypes.GasPriceChain) error {
 	return nil
 }
 
+// validateMerkleRootsState merkle roots seq nums validation by comparing with on-chain state.
+func validateMerkleRootsState(
+	ctx context.Context,
+	lggr logger.Logger,
+	report cciptypes.CommitPluginReport,
+	reader cciptypes.CCIPReader,
+) (bool, error) {
+	reportChains := make([]cciptypes.ChainSelector, 0)
+	reportMinSeqNums := make(map[cciptypes.ChainSelector]cciptypes.SeqNum)
+	for _, mr := range report.MerkleRoots {
+		reportChains = append(reportChains, mr.ChainSel)
+		reportMinSeqNums[mr.ChainSel] = mr.SeqNumsRange.Start()
+	}
+
+	if len(reportChains) == 0 {
+		return true, nil
+	}
+
+	onchainSeqNums, err := reader.NextSeqNum(ctx, reportChains)
```

Comment:
Lets disambiguate and make it clear which sequence numbers we're talking about


```suggestion
	onchainNextExpectedSeqNums, err := reader.NextSeqNum(ctx, reportChains)
```

Reply:
I'll rename the interface method, contract returns the last committed seq num. And rename var to `onchainCommittedSeqNums`

Reply:
https://github.com/smartcontractkit/chainlink-common/pull/627